assets/src/edit-story/components/library/panes/media/media3p/api/media3pApiProvider.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { createResource } from '../../../../../../app/media/utils';
+import { listMedia as apiListMedia } from './apiFetcher';
+import Context from './context';
+
+/**
+ * The supported providers.
+ *
+ * @enum {string}
+ */
+const Providers = {
+  UNSPLASH: 'unsplash',
+};
+
+/**
+ * Provider for the Media3P API. Delegates fetching the data to apiFetcher,
+ * but transforms the response into resources.
+ */
+function Media3pApiProvider({ children }) {
+  const MEDIA_PAGE_SIZE = 20;
+
+  function constructFilter(provider, searchTerm, mediaType) {
+    let tokens = [];
+    if (provider) {
+      tokens.push(`provider:${provider}`);
+    }
+    if (mediaType) {
+      tokens.push(`type:${mediaType}`);
+    }
+    if (searchTerm) {
+      tokens.push(searchTerm);
+    }
+    return tokens.join(' ');
```

Comment:
Optional, does this work?

return [
  provider ? `provider:${provider} : null,
  mediaType ? `type:${mediaType} : null,
  searchTerm
].filter(Boolean).join(' ')

Reply:
Nice!